### PR TITLE
adding csharp example

### PR DESCRIPTION
**Pull request by neilwashere**

### CODE REVIEW

File: csharp/src/Program.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                .WithParsed(options =>
+                {
+                    var token = Sign(options.Kid, options.KeyFile, options.PayloadFile);
+                    Console.Write(token);
+                });
+        }
+
+        static string Sign(string kid, string keyFile, string payloadFile)
+        {
+            // THe base64 encoded content of the ec private key (--BEGIN..--/--END..-- removed)
```
Comment (LukeMathWalker) [inline csharp/src/Program.cs:26]:
```suggestion
            // The base64 encoded content of the ec private key (--BEGIN..--/--END..-- removed)
```

File: csharp/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+## Example
+
+This assumes you have generated a key pair using the `generate_key_pair.sh` script in the root folder of the project.
+
+```bash
+# Assumes you are running from within the `csharp` directory
+dotnet run --project src/signing.csproj -- --kid $kid --key-file ./ec512-private-key.pem --payload ./payload.json
+
+# slurp into shell variable
+signature=$(dotnet run --project src/signing.csproj -- --kid $kid --key-file ./ec512-private-key.pem --payload ./payload.json 2>/dev/null)
```
Comment (tl-alex-butler) [inline csharp/README.md:23]: From `src/signing.csproj` I assume you're meant to run these from the _csharp_ dir.

```suggestion
dotnet run --project src/signing.csproj -- --kid $kid --key-file ../ec512-private-key.pem --payload ../payload.json

# slurp into shell variable
signature=$(dotnet run --project src/signing.csproj -- --kid $kid --key-file ../ec512-private-key.pem --payload ../payload.json 2>/dev/null)
```

File: csharp/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+## Troubleshooting
+Dangling control chars in the source file can cause validation issues. Run with `--debug` to see the bits *actually* slurped in.
+
+```bash
+dotnet run --project src/signing.csproj -- --kid $kid --key-file ./ec512-private-key.pem --payload ./payload.json --debug
```
Comment (tl-alex-butler) [inline csharp/README.md:30]:
```suggestion
dotnet run --project src/signing.csproj -- --kid $kid --key-file ../ec512-private-key.pem --payload ../payload.json --debug
```

**Review by tl-alex-butler — DISMISSED** [review]
this works for me, nice and simple :medal_sports:

File: csharp/README.md (hunk 2 above)
Comment (neilwashere) [reply csharp/README.md:23]: I actually ran these from within the csharp directory and copied my assets there. Taking from the parent is probably more consistent though :+1:

**Pull request merged**